### PR TITLE
Sinch - Huawei OAuth flow for Huawei Push Notifications

### PR DESCRIPTION
**Pull request by christoffer-a-sinch**
RTC-5789

### CODE REVIEW

File: docs/voice/voice-for-android-cloud/huawei-oauth.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+1. Make use of the _Huawei_ (_HMS_) APIs in the Sinch Android SDK, see details [here](TODO INSERT LINK).
+2. Implement an _OAuth 2.0 Authorization Server_ endpoint that can provide Sinch with _OAuth 2.0_ _access_tokens_ required to send messages via _HPK_ on your behalf.
+
+## Huawei OAuth Flow
+
+Sinch will send push notification messages via the [_HPK_ API](https://developer.huawei.com/consumer/en/doc/HMSCore-References-V5/https-send-api-0000001050986197-V5). Huawei Push Kit supports (and require) an _OAuth 2.0_ _Client Credentials_ flow to authenticate against the _Huawei Push Kit_ server endpoint(s). Your Huawei App in Huawei _AppGallery Connect_ will have an _App ID_ and an _App secret_. These are to be used as OAuth client credentials, i.e. `client_id` and `client_secret` respectively.
+
```
Comment (reddvl1980) [inline docs/voice/voice-for-android-cloud/huawei-oauth.md:11]: I think we need to clarify here what `key` and `id` we are talking about.
On Huawei AppGallery Connect apps are organized in projects. Projects have `client id` and `client key`, while applications have `app id` and `app key` and `API key`. I'm not 100% sure, what's used where/when.
Screenshot 2020-09-25 at 14 21 48

File: docs/voice/voice-for-android-cloud/huawei-oauth.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Sinch supports Huawei OAuth flow by delegation. You will keep your `client_secret` on your backend, and Sinch will request an OAuth `access_token` via an _Authorization Server_ endpoint that you implement. The overall flow is depicted below:
+
+![Sinch - Huawei OAuth flow](images\20200922-sinch-huawei-hpk-oauth.png)
+
+Key takeaways:
+
+1. When Sinch need an `access_token` required to send a push message to _Huawei Push Kit_ server, it will make an OAuth request using a _Client Credentials_ grant type to your _Authorization Server_. This request will be specifying and `client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer` will as value for `client_assertion` provide a JWT that is symmetrically signed with your _Sinch Application Secret_.
+2. Your _Authorization Server_ should validate the JWT provided as `client_assertion` by Sinch and that the signed JWT is signed with your _Sinch Application Secret_.
```
Comment (reddvl1980) [inline docs/voice/voice-for-android-cloud/huawei-oauth.md:19]: I think it worth mention here about expiry somehow. E.g. clarify, that Sinch DOES NOT need an `access token` each time somebody's making a call.

File: docs/voice/voice-for-android-cloud/huawei-oauth.md (hunk 1 above)
Comment (christoffer-a-sinch) [reply docs/voice/voice-for-android-cloud/huawei-oauth.md]: I've verified it is "App ID" and "App secret" that are to be used as OAuth `client_id` and `client_secret` respectively. API key and Client key are for other uses.

Comment (reddvl1980) [reply docs/voice/voice-for-android-cloud/huawei-oauth.md]: good

**Pull request merged**